test/unit/transitions/multi_form_alerts.js

```diff
@@ -0,0 +1,384 @@
+const _ = require('underscore'),
+      config = require('../../../config'),
+      lineage = require('../../../lib/lineage'),
+      messages = require('../../../lib/messages'),
+      sinon = require('sinon').sandbox.create(),
+      transition = require('../../../transitions/multi_form_alerts'),
+      utils = require('../../../lib/utils'),
+      vm = require('vm');
+
+exports.tearDown = callback => {
+  sinon.restore();
+  callback();
+};
+
+exports.setUp = callback => {
+  sinon.stub(utils, 'getReportsWithinTimeWindow').returns(Promise.resolve(reports));
+  sinon.stub(lineage, 'fetchHydratedDocPromise').callsFake(id => {
+    if (id === doc._id) { return Promise.resolve(doc); }
+    if (id === reports[0]._id) { return Promise.resolve(hydratedReports[0]); }
+    if (id === reports[1]._id) { return Promise.resolve(hydratedReports[1]); }
+  });
```

Comment:
The new function is really unclear too. I'll change it and show you what I mean.